client/src/pages/CommunityPage/CommunityPageCreateGroup.js

```diff
@@ -0,0 +1,194 @@
+import React, { Component } from "react";
+import "./CommunityPageCreateGroup.css";
+import Banner from "../../components/Banner/Banner";
+import InputTextField from "../../components/InputTextField/InputTextField";
+import TextField from "@material-ui/core/TextField";
+import Banner__Community from "assets/images/community_banner.png";
+import Button from "../../components/Button/Button";
+import axios from "axios";
+import { BASE_URL } from "../../services/constants";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+export default class CommunityPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: "",
+      description: "",
+      location: null,
+    };
+  }
+
+  handleChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
+  submit = () => {
+    console.log(this.state);
+    axios({
+      method: 'POST',
+      url: `${BASE_URL}/groups`,
+      data: {
+        title: this.state.title,
+        description: this.state.description,
+        location: this.state.location
+      }
+    })
+  };
+
+  componentDidMount() {
+    this.getUSStates();
+  }
+
+  getUSStates = () => {
+    axios.get(`${BASE_URL}/location/states`).then(res => {
+      console.log(res.data);
+      this.setState({ USStates: res.data });
+    });
+  };
+
+  getCitiesFromState = event => {
+    if (event.target.textContent === "") this.setState({ stateFound: false });
+
+    let USStates = this.state.USStates;
+
+    //TextContent because materialUI outputs an <li> tag
+    let userSubmission = event.target.textContent.toLowerCase();
+
+    USStates.forEach(state => {
+      if (state.name.toLowerCase() === userSubmission) {
+        axios.get(`${BASE_URL}/location/bystate?state__code=${state.code}`).then(res => {
+          this.setState({
+            stateFound: true,
+            currentStateCities: res.data
+          });
+        });
+      }
+    });
+  };
```

Comment:
This is going to be a lot of ajax requests - maybe we can load the cities once the state is selected, like we do on the account creation flow?

Reply:
I used the account creation flow to do this part. I thought that with the conditional there it will wait to find a match for a valid state before loading the cities.